mailbox/api/src/main/java/org/apache/james/mailbox/model/QuotaOperation.java

```diff
@@ -0,0 +1,71 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.model;
+
+import java.util.Objects;
+
+import org.apache.james.core.quota.QuotaCountUsage;
+import org.apache.james.core.quota.QuotaSizeUsage;
+
+import com.google.common.base.Preconditions;
+
+public class QuotaOperation {
+    private final QuotaRoot quotaRoot;
+    private final QuotaCountUsage count;
+    private final QuotaSizeUsage size;
+
+    public QuotaOperation(QuotaRoot quotaRoot, QuotaCountUsage count, QuotaSizeUsage size) {
+        Preconditions.checkArgument(count.asLong() > 0, "Count should be positive");
+        Preconditions.checkArgument(size.asLong() > 0, "Size should be positive");
```

Comment:
We miss the 'strictly' in the message

Reply:
Why can't it be equals to 0 ?

Reply:
looks like we can only change both at the same time but not `count` or `size` only

Reply:
> Why can't it be equals to 0 ?

Well from what I observed:
* that was already the case in the code before I made any changes
* the test suite already present makes sure of that as well (but because it's enforced in the pojo then I moved those in the pojo tests)
* the reason for this is then I believe because we never call increase or decrease if those are at 0. What would be the point to do an operation on increasing or decreasing the quota on 0? Unnecessary operation right?